ocaml/testsuite/tests/pattern-guards/test.ml

```diff
@@ -58,6 +58,17 @@ Error: This pattern matches values of type 'a option
        but a pattern was expected which matches values of type int
 |}]
 
```

Comment:
Curious: do you get the "total match" warning if the scrutinee `f x` is an uninhabited type? (Should you? I don't have too strong of a position either way.)

Reply:
You do not, I added a test demonstrating this! I think you would need a case that looks like `| _ -> .` for the typechecker to conclude that the cases are total.